ggshield/utils/git_shell.py

```diff
@@ -95,6 +96,25 @@ def _git_rev_parse(option: str, wd: Path) -> Optional[str]:
     return _git_rev_parse_absolute(option=option, wd_absolute=wd.resolve())
 
 
+def _parse_from_git_url(url: str) -> str:
```

Comment:
I think we should have explicit test for this function with many cases.

Reply:
I've moved the test urls from `test_scan_context` to the new test for this function, because I don't think testing all the urls twice is necessary. Do you agree?